### PR TITLE
Fix for Issue #1 and some other enhancements

### PR DESCRIPTION
**Pull request by tprochazka**
I did this changes:
- fixed bug #1 (Editor return original editor instead secured one in fluent API
- added support for setStringSet() and getStringSet() method introduced in API 11 but in compatible way with API 8 (own implementation)
- added support for apply() introduced in API 9 with workaround for API 8

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -31,15 +31,25 @@
...
 	</scm>
 	<distributionManagement>
-		<site>
-			<id>starhost-andsecureprefs</id>
-			<url>ftp://starhost.hu/web/andsecureprefs/</url>
-		</site>
+	<repository>
+		<id>nexus</id>
+		<name>Inmite 3rd party Repo</name>
+		<url>http://mvn.office.inmite.eu/nexus/content/repositories/thirdparty</url>
```
Comment (kovmarci86) [inline pom.xml]: mvn.office.inmite.eu is not reachable from outside

**Comment by kovmarci86** [comment]
Hi tprochazka,
Thank you for the pull-request! I'm happy to see someone's contribution.
The most of the changes are reasonable and great, but there are some issues.
The repositories are not reachable from outside of your company therefore it must not be there.
For the SecuredEditor's builder method return statements: you were correct, it returned the wrong editor, but I'm not sure it is good to return the SecuredEditor, as the main concept is to give completely the same interface than SharedPreferences. The SecuredEditor is a public class therefore it must not contain any extra public methods to avoid any kind of dependency to this library. (it can be plugged in easily, but also can be removed if someone decides not to use it)
This is the case for the edit method of SecureSharedPreferences class.
What do you think about this?
Thanks again, best regards,
Marcell

**Comment by tprochazka** [comment]
I'm sorry, repository was added by mistake. I removed it.

I also agree with you second objection, but there is not possible because I added support for methods which was introduced in newer version that API 8, like apply() or setStringSet(). These methods are implemented to work also on API 8. So you can write the same code that will work on API 8 and API 17. If SecureEditor will return only Editor it will not work on old Android API because some methods are missing  here. You can try it.

**Comment by tprochazka** [comment]
So, what do you mean?

**Pull request merged**